ingestion_program/systematics.py

```diff
@@ -598,19 +598,24 @@ def get_bootstrapped_dataset(
     Seed = seed
     for i, key in enumerate(test_set.keys()):
         Seed = Seed + i
-        weights = test_set[key].pop("weights")
+
+        temp_data = test_set[key].copy()
+
+        weights = temp_data["weights"].copy()
```

Comment:
Not sure we need to copy here, I don't thing we are writing to `weights`?

Reply:
in line 604 yes the copy seem unnecessary. But for line 602  we need it right ?

Reply:
No, I don't thing so it can be removed. I think the following would be more efficient:

```python
temp_data = temp_data[new_weights > 0] # This will already create a new array
temp_data["weights"] = new_weights
```

Reply:
so only use temp array at the last stage where we reassign the weights right

```python
    pseudo_data = []
    Seed = seed
    for i, key in enumerate(test_set.keys()):
        Seed = Seed + i

        if poisson:
            random_state = np.random.RandomState(seed=Seed)
            new_weights = random_state.poisson(bkg_norm[key] * test_set[key]["weights"])
        else:
            new_weights = bkg_norm[key] * test_set[key]["weights"]

        temp_data = test_set[key][new_weights > 0]

        pseudo_data.append(temp_data)

    pseudo_data = pd.concat(pseudo_data)
    
```

Reply:
Yes, so you still need
```
temp_data["weights"] = new_weights
```

before the append?

Reply:
Oh yes I missed that.